airflow/providers/microsoft/azure/hooks/base_azure.py

```diff
@@ -0,0 +1,68 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from os import getenv
+from typing import Any
+
+from azure.common.client_factory import get_client_from_auth_file
+from azure.common.credentials import ServicePrincipalCredentials
+
+from airflow import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+class AzureBaseHook(BaseHook):
+    """
+    This hook acts as a base hook for azure services. It offers several authentication mechanisms to
+    authenticate the client library used for upstream azure hooks.
+
+    :param sdk_client: The SDKClient to use.
+    :param conn_id: The azure connection id which refers to the information to connect to the service.
+    """
+
+    def __init__(self, sdk_client: Any, conn_id: str = 'azure_default'):
+        self.sdk_client = sdk_client
+        self.conn_id = conn_id
+        super().__init__()
+
+    def get_conn(self) -> Any:
+        """
+        Authenticates the resource using the connection id passed during init.
+
+        :return: the authenticated client.
+        """
+        conn = self.get_connection(self.conn_id)
+
+        key_path = conn.extra_dejson.get('key_path') or getenv('AZURE_AUTH_LOCATION')
```

Comment:
Should we support key as dict? This will facilitate use with secret backends.

Reply:
Makes sense - will add that. 👍

Reply:
good point

Reply:
By the way..
```suggestion
        key_path = conn.extra_dejson.get('key_path')
```
works also, because azure will check the env variable, too.
>     auth_path = auth_path or os.environ['AZURE_AUTH_LOCATION']

Reply:
So I am removing also the test case because the functionality is being provided (and should be tested) by azure.